src/components/Modal.tsx

```diff
@@ -36,14 +37,25 @@ class Modal extends React.PureComponent<ModalProps> {
       this.node = document.createElement("div");
     }
 
-    return visible
-      ? ReactDOM.createPortal(
-          <div className={cx(modalContainer, "ReactPortal", className)}>
-            {children}
-          </div>,
-          this.node
-        )
-      : null;
+    return ReactDOM.createPortal(
+      <MountTransition visible={visible}>
+        {transitionStyles => (
+          <div
+            style={{
+              opacity: transitionStyles.opacity
+            }}
+            className={cx(modalContainer, "ReactPortal", className)}
+          >
+            <div
+              style={{ transform: transitionStyles.transform, display: "flex" }}
```

Comment:
1. The first div is the overlay and the second div is the content in the middle.
2. Everything by default uses flex so I kept it there so that things don't break. else Confirmation dialogue was breaking so I expect other things to break.